### PR TITLE
[Merged by Bors] - refactor(NumberTheory/FLT): Define Fermat's Last Theorem for fixed exponent

### PR DESCRIPTION
**Pull request by tb65536**
This PR adds a definition of Fermat's Last Theorem for fixed exponent. The motivation for this is that `FermatLastTheoremWith ℕ n`, `FermatLastTheoremWith ℤ n`, and `FermatLastTheoremWith ℚ n` are all equivalent, so it would be nice to have a canonical name, rather than sometimes referring to one and sometimes referring to another.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
It looks a good idea to me. Since you are at it, can you add that it is false for `n = 0, 1, 2`?

File: Mathlib/NumberTheory/FLT/Basic.lean [hunk 1]
```diff
@@ -28,6 +28,15 @@ def FermatLastTheoremFor (n : ℕ) : Prop := FermatLastTheoremWith ℕ n
 when `n ≥ 3`. -/
 def FermatLastTheorem : Prop := ∀ n ≥ 3, FermatLastTheoremFor n
 
+lemma fermatLastTheorem_zero : FermatLastTheoremFor 0 :=
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/FLT/Basic.lean:31]: Ahahah, it is true for `0` :D

Comment (tb65536) [reply Mathlib/NumberTheory/FLT/Basic.lean:31]: Yeah, surprised me too!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**